lib/node_modules/@stdlib/ndarray/base/nullary/test/test.3d.js

```diff
@@ -201,6 +257,96 @@ tape( 'the function applies a nullary callback to each indexed element of a 3-di
 	t.end();
 });
 
+tape( 'the function applies a nullary callback to each indexed element of a 3-dimensional ndarray (row-major, non-contiguous, large arrays)', function test( t ) {
+	var expected;
+	var bsize;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+
+	dt = 'float64';
+	ord = 'row-major';
+
+	bsize = blockSize( dt );
+	sh = [ bsize*2, 1, 2 ];
+	st = [ -4, -4, 2 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, zeros( numel( sh )*2, dt ), sh, st, o, ord );
+
+	nullary( [ x ], constantFunction( 10.0 ) );
+
+	expected = new Float64Array( x.length*2 );
+	dfill( x.length, 10.0, expected, st[ 2 ] );
+
+	t.strictEqual( isSameFloat64Array( x.data, expected ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function applies a nullary callback to each indexed element of a 3-dimensional ndarray (row-major, non-contiguous, large arrays)', function test( t ) {
+	var expected;
+	var bsize;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+
+	dt = 'float64';
+	ord = 'row-major';
+
+	bsize = blockSize( dt );
+	sh = [ 1, bsize*2, 2 ];
+	st = [ -4, -4, 2 ];
```

Comment:
These strides seem off. I believe this should be

```suggestion
	st = [ -bsize*4, -4, 2 ];
```

Consider, e.g., 

```
In [22]: shape2strides( [ 1, 64*2, 2 ], 'row-major' )
Out[22]: [ 256, 2, 1 ]
```

What you're doing here is effectively iterating overlapping segments by only have `-4` as your first stride. The tests pass, but only because `sh[0] === 1`.